src/App.ts

```diff
@@ -795,174 +760,173 @@ const validViewTypes = ['view_closed', 'view_submission'];
-        (body as SlackEventMiddlewareArgs['body']).authorizations![0].enterprise_id !== null
-          ? ((body as SlackEventMiddlewareArgs['body']).authorizations![0].enterprise_id as string)
-          : type === IncomingEventType.Event || type === IncomingEventType.Command
-          ? ((body as (SlackEventMiddlewareArgs | SlackCommandMiddlewareArgs)['body']).enterprise_id as string)
-          : (type === IncomingEventType.Action ||
-              type === IncomingEventType.Options ||
-              type === IncomingEventType.ViewAction ||
-              type === IncomingEventType.Shortcut) &&
-            body.team !== null
-          ? ((body as (
-              | SlackActionMiddlewareArgs
-              | SlackOptionsMiddlewareArgs
-              | SlackViewMiddlewareArgs
-              | SlackShortcutMiddlewareArgs
-            )['body']).team!.enterprise_id as string)
-          : (type === IncomingEventType.Action ||
-              type === IncomingEventType.Options ||
-              type === IncomingEventType.ViewAction ||
-              type === IncomingEventType.Shortcut) &&
-            body.enterprise !== undefined
-          ? ((body as (
-              | SlackActionMiddlewareArgs
-              | SlackOptionsMiddlewareArgs
-              | SlackViewMiddlewareArgs
-              | SlackShortcutMiddlewareArgs
-            )['body']).enterprise!.id as string)
-          : undefined,
-      userId:
-        type === IncomingEventType.Event
-          ? typeof (body as SlackEventMiddlewareArgs['body']).event.user === 'string'
-            ? ((body as SlackEventMiddlewareArgs['body']).event.user as string)
-            : typeof (body as SlackEventMiddlewareArgs['body']).event.user === 'object'
-            ? ((body as SlackEventMiddlewareArgs['body']).event.user.id as string)
-            : (body as SlackEventMiddlewareArgs['body']).event.channel !== undefined &&
-              (body as SlackEventMiddlewareArgs['body']).event.channel.creator !== undefined
-            ? ((body as SlackEventMiddlewareArgs['body']).event.channel.creator as string)
-            : (body as SlackEventMiddlewareArgs['body']).event.subteam !== undefined &&
-              (body as SlackEventMiddlewareArgs['body']).event.subteam.created_by !== undefined
-            ? ((body as SlackEventMiddlewareArgs['body']).event.subteam.created_by as string)
-            : undefined
-          : type === IncomingEventType.Action ||
-            type === IncomingEventType.Options ||
-            type === IncomingEventType.ViewAction ||
-            type === IncomingEventType.Shortcut
-          ? ((body as (SlackActionMiddlewareArgs | SlackOptionsMiddlewareArgs | SlackViewMiddlewareArgs)['body']).user
-              .id as string)
-          : type === IncomingEventType.Command
-          ? ((body as SlackCommandMiddlewareArgs['body']).user_id as string)
-          : undefined,
-      conversationId: channelId,
-      isEnterpriseInstall: false,
-    };
+
```

Comment:
This refactor is mostly borrowed from #669, so that when it eventually lands, there's less chance of a merge conflict.

I found all the Org Apps related changes on `main` and layered them into this second of code from that refactor.

Prettier undid some of my style related changes, but these will likely be automatically fixed again once the linked PR lands.

Reply:
I'm a big fan of this refactor. Much easier to read and follow